### DISCUSSION TITLE
Valid function combination (usageFlags) for single timer

### DISCUSSION THREAD

**Discussion by Strat00s** [iNavFlight/inav#8247] [Q&A]
Hey everyone. I am currently working on a script for remapping servos and motors on a fixed-wing. And I have a question about functions (`usageFlag`) that can be used at the same time on the same timer.

From what I found, the biggest problems are servos and motors using the same timer, as they usually run at different frequencies (although it seems that they get remapped in `pwm_mapping.c`, so it probably isn't a problem, but it might not have the desired function).

So, from the list below, are there any other functions that shouldn't or can't be used together on the same timer?
```
typedef enum {
    TIM_USE_ANY             = 0,
    TIM_USE_PPM             = (1 << 0),
    TIM_USE_PWM             = (1 << 1),
    TIM_USE_MC_MOTOR        = (1 << 2),     // Multicopter motor output
    TIM_USE_MC_SERVO        = (1 << 3),     // Multicopter servo output (i.e. TRI)
    TIM_USE_MC_CHNFW        = (1 << 4),     // Deprecated and not used after removal of CHANNEL_FORWARDING feature
    TIM_USE_FW_MOTOR        = (1 << 5),
    TIM_USE_FW_SERVO        = (1 << 6),
    TIM_USE_LED             = (1 << 24),
    TIM_USE_BEEPER          = (1 << 25),
} timerUsageFlag_e;
```

Edit: I might as well ask about DMA. How does one determine whether to set the `dmavar` to 0 or 1 in `DEF_TIM()` macro?

**Comment by DzikuVx** [comment]
No, it's not possible to mix TIM_USE_MC_MOTOR and TIM_USE_MC_SERVO or TIM_USE_FW_MOTOR and TIM_USE_FW_SERVO  on one timer. One timer can work with one frequency

**Reply by Strat00s** [reply]
Well yes, I know. But are there any other combinations that should be avoided?

**Reply by DzikuVx** [reply]
No, not really

**Reply by Strat00s** [reply]
Thanks for the reply. And do you have any insight on the `dmavar` and when it should be enabled?
I found some [datasheets with DMA streams and timer/channel mappings](https://www.st.com/resource/en/application_note/dm00046011-using-the-stm32f2-stm32f4-and-stm32f7-series-dma-controller-stmicroelectronics.pdf), but I don't really know what (and when) is the DMA used for in iNav.
